### DISCUSSION TITLE
frb_workerpool crashes on Linux 6.9.3-arch1-1

### DISCUSSION THREAD

**Discussion by rustdesk** [fzyzcjy/flutter_rust_bridge#2081] [General]
We are experiencing a crash without a precise pattern of use. Could you help give some tip?

brige version: 1.80.1
rust version: 1.78
tokio version: 1.38

https://github.com/rustdesk/rustdesk/issues/8320

> flutter: [Remote Page] call get_remote_list with args null from window 1
> thread 'frb_workerpool' panicked at library/std/src/sys_common/process.rs:155:17:
> called Result::unwrap() on an Err value: Os { code: 9, kind: Uncategorized, message: "Descrittore di file errato" }
> stack backtrace:
> note: Some details are omitted, run with RUST_BACKTRACE=full for a verbose backtrace.
> [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: FfiException(PANIC_ERROR, called Result::unwrap() on an Err value: Os { code: 9, kind: Uncategorized, message: "Descrittore di file errato" }, null)
> #0 FlutterRustBridgeBase._transformRust2DartMessage (package:flutter_rust_bridge/src/basic.dart:129:9)
> https://github.com/rustdesk/rustdesk/issues/1 FlutterRustBridgeBase.executeNormal. (package:flutter_rust_bridge/src/basic.dart:70:9)
> 
> https://github.com/rustdesk/rustdesk/issues/2 _DesktopHomePageState.initState. (package:flutter_hbb/desktop/pages/desktop_home_page.dart:669:21)
> 
> https://github.com/rustdesk/rustdesk/issues/3 periodic_immediate. (package:flutter_hbb/common.dart:2404:5)
> Gdk-Message: 15:14:28.597: Error 32 (Pipe interrotta) dispatching to Wayland display.

**Comment by fzyzcjy** [comment]
Hi, it would be great to have a Rust stack trace, then we can know what exactly place it really panics.

In v2, it is enabled by default; but in v1, maybe need some configuration https://cjycode.com/flutter_rust_bridge/guides/how-to/stack-trace.

By the way, I personally suggest migrating to v2. (It looks like ".dev" version but is mainly because I want to allow to have seldom breaking changes)

**Reply by rustdesk** [reply]
Thanks, It happened on one user's machine only.  I will build an debug version for him.
Migrating bridge is not an easy job. That's why we are still stick to 1.80.1

**Reply by fzyzcjy** [reply]
I see. Ping me if there are any new clues!

Btw, I personally enable backtrace also for release build, because this (roughly) does not harm performance if there is no errors, and helps a lot when error happens.
